mutant/modules/sarscov2_report.py

```diff
@@ -77,22 +78,28 @@ def create_trailblazer_config(self) -> None:
             yaml.dump(data={"jobs": finished_slurm_ids}, stream=trailblazer_config_file)
 
     def create_concat_pangolin(self):
-        indir = "{0}/ncovIllumina_sequenceAnalysis_pangolinTyping".format(self.indir)
+        """Concatenate pangolin results"""
 
-        concat = open("{0}/{1}.pangolin.csv".format(self.indir, self.ticket), "w+")
+        indir = "{0}/ncovIllumina_sequenceAnalysis_pangolinTyping".format(self.indir)
+        concatfile = "{0}/{1}.pangolin.csv".format(self.indir, self.ticket)
         pangolins = glob.glob("{0}/*.pangolin.csv".format(indir))
-        with open(pangolins[0], "r") as f:
-            header = f.readlines()[0]
-        concat.write(header)
-        for item in pangolins:
-            single = open(item, "r")
-            concat.write("\n".join(single.readlines()[1:]))
-        concat.close()
+        # Copy header
+        header = read_filelines(pangolins[0])[0]
+        with open(concatfile, "w") as concat:
+            concat.write(header)
+            # Parse sample pangolin data
+            for pango in pangolins:
+                data = read_filelines(pango)[1:]
+                for line in data:
+                    # Use sample name at taxon field
+                    sample = line.split(",")[0].split(".")[0].split("_")[-1]
```

Comment:
This would be much safer as a regex re.sub or similar.

Reply:
Sure, I'll have a look at it!